frontends/common/options.cpp

```diff
@@ -130,6 +133,38 @@ void CompilerOptions::setInputFile() {
     }
 }
 
+std::vector<const char*>* CompilerOptions::process(int argc, char* const argv[]) {
+    char buffer[PATH_MAX];
+    int len;
+    struct stat st;
+    if ((len = readlink("/proc/self/exe", buffer, sizeof(buffer))) > 0 ||
+        (len = readlink("/proc/curproc/exe", buffer, sizeof(buffer))) > 0 ||
+        (len = readlink("/proc/curproc/file", buffer, sizeof(buffer))) > 0 ||
+        (len = readlink("/proc/self/path/a.out", buffer, sizeof(buffer))) > 0) {
+        buffer[len] = 0;
+    } else if (argv[0][0] == '/') {
+        strcpy(buffer, argv[0]);
+    } else if (strchr(argv[0], '/')) {
+        getcwd(buffer, sizeof(buffer));
+        strncat(buffer, argv[0], sizeof(buffer) - strlen(buffer) - 1);
+    } else if (getenv("_")) {
```

Comment:
Some shells (bash, ksh, and zsh) set the `_` environment variable to the full path of the binary they're about to execute.